lib/use-public-interface.js

```diff
@@ -35,32 +22,61 @@ export const usePublicInterface = ({ host, visibleData, ...api }) => {
 				return removed[0];
 			}
 		},
-		replaceItem = (oldItem, newItem) => {
-			const itemIndex = host.data.indexOf(oldItem);
-			if (itemIndex > -1) {
-				return host.replaceItemAtIndex(itemIndex, newItem);
-			}
-		},
 		replaceItemAtIndex = (index, newItem) => {
 			host.data.splice(index, 1, newItem);
 			host.data = host.data.slice();
+		};
+
+	return {
+		removeItem,
+		removeItems(items) {
+			const removedItems = [];
+
+			for (let i = items.length - 1; i >= 0; i -= 1) {
+				const removed = host.removeItem(items[i]);
+				if (removed != null) {
+					removedItems.push(removed);
+				}
+			}
+			return removedItems;
 		},
-		selectItem = (item) => {
+
+		replaceItemAtIndex,
+		replaceItem(oldItem, newItem) {
+			const itemIndex = host.data.indexOf(oldItem);
+			if (itemIndex > -1) {
+				return replaceItemAtIndex(itemIndex, newItem);
+			}
+		},
+
+		selectItem(item) {
 			host.shadowRoot.querySelector('#groupedList').select(item);
 		},
-		selectAll = () => {
+		selectAll() {
 			host.shadowRoot.querySelector('#groupedList').selectAll();
 		},
-		deselectAll = () => {
+		deselectAll() {
 			host.shadowRoot.querySelector('#groupedList').deselectAll();
 		},
-		deselectItem = (item) => {
+		deselectItem(item) {
 			host.shadowRoot.querySelector('#groupedList').deselect(item);
 		},
-		isItemSelected = (item) => {
+		isItemSelected(item) {
 			return host.shadowRoot.querySelector('#groupedList').isItemSelected(item);
 		},
-		{ setFilterState } = api;
+	};
+};
+
+export const usePublicInterface = ({
+	host,
+	visibleData,
+	...api
+}) => {
+	const { setFilterState } = api,
+		napi = useMemo(() => mkNapi(host), []);
+
+	useImperativeApi(api, Object.values(api));
+	useImperativeApi(napi, Object.values(napi));
```

Comment:
```suggestion
	useImperativeApi(napi, []);
```
Because napi values never change

Reply:
No. This makes it clear that napi would be re-applied when its values change. The values will never change because of the useMemo above but that is a separate thing.